x-pack/plugins/observability_solution/apm/server/utils/parse_es_filters_or_throw.test.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { parseEsFiltersOrThrow } from './parse_es_filters_or_throw';
+
+describe('parseEsFiltersOrThrow', () => {
+  it('should parse must_not clause', () => {
+    const filters = JSON.stringify({
+      must_not: [{ term: { 'service.name': 'opbeans-node' } }],
+    });
+    expect(parseEsFiltersOrThrow(filters)).toEqual({
+      should: [],
+      must: [],
+      must_not: [{ term: { 'service.name': 'opbeans-node' } }],
+      filter: [],
+    });
+  });
+
+  it('should parse filters clause', () => {
+    const filters = JSON.stringify({
+      filter: [{ term: { 'service.name': 'opbeans-node' } }],
+    });
+    expect(parseEsFiltersOrThrow(filters)).toEqual({
+      should: [],
+      must: [],
+      filter: [{ term: { 'service.name': 'opbeans-node' } }],
+      must_not: [],
+    });
+  });
+
+  it('throws on invalid filters caluse', () => {
+    const filters = JSON.stringify({
+      filter: { term: { 'service.name': 'opbeans-node' } },
+    });
+    expect(() => parseEsFiltersOrThrow(filters)).toThrow(
+      'Failed to parse filters: filter is not iterable'
+    );
+  });
+
+  it('throws on invalid must_not caluse', () => {
+    const filters = JSON.stringify({
+      must_not: { term: { 'service.name': 'opbeans-node' } },
+    });
+    expect(() => parseEsFiltersOrThrow(filters)).toThrow(
+      'Failed to parse filters: must_not is not iterable'
+    );
+  });
+
+  it('does not support should or must clauses', () => {
+    const filters = JSON.stringify({
+      should: [{ term: { 'service.name': 'opbeans-node' } }],
+      must: [{ term: { 'service.name': 'opbeans-node' } }],
+    });
```

Comment:
Following the io-ts comment above. The custom io-ts you created could validate if the client has defined these two fields and throw an error since it's not yet supported.

Reply:
I had updated the runtime type, but now that I think about it I'm not gonna throw an error. The reason is that the `buildQueryFromFilters` helper which is used internally to build filters correctly actually returns `must` and `should` as empty arrays.